lib/ansible/galaxy/role.py

```diff
@@ -379,7 +379,11 @@ def install(self):
                                     if n_part != '..' and not n_part.startswith('~') and '$' not in n_part:
                                         n_final_parts.append(n_part)
                                 member.name = os.path.join(*n_final_parts)
-                                role_tar_file.extract(member, to_native(self.path))
+                                if hasattr(tarfile, 'data_filter'):
+                                    # Remove this check when Python 3.11 is EOL
```

Comment:
Use a deprecation comment here (implemented by our own `deprecated` plugin for `pylint`):

```suggestion
                                    # deprecated: description='extract fallback without filter' python_version='3.11'
```